### ISSUE TITLE
Confused with the design

### ISSUE THREAD

**Issue by Deexaa** [MasLiang/Learning-without-Forgetting-using-Pytorch#12]
Thanks for sharing your code!
But I have a little questions about the design of incremental learning. In the original paper of Lwf, it seems that the author add different datasets to the model as a new task, so they have 2 tasks and 2 datasets. In the public discussions, I saw people add different classes in the same dataset as a new task, so they have 2+ tasks(like 2+2+2) and 1 datasets(6 classes totally). But in your codes, it seems that you add one classes in the same dataset as a new task, so you have 1+ tasks(like 3+1+1) and 1 datasets(6 classes totally). I am confused that whether these 3 settings are all called class-incremental? If not, which is the right one.
I'd really appreciate it if you could answer my questions! Thanks a lot!

**Comment by MasLiang** [comment]
Firstly, you need to know what is dataset. A dataset is a set of data. You only think CIFAR-10 is a dataset, a ISLVRC2012 is a dataset. However, you can also merge these two dataset. It is also a new dataset. And you can also divide some ISLVRC2012 to two parts and call them two new dataset. So you do not need to care the additional classes belongs to which datasets I think. Of course, classes in the narrow dataset have some attribute such as the have the same size, they are mutually exclusive (if you believe these authors) and so on. So, it is true that using new classes in the same narrow dataset might  be different with using classes form other narrow datasets. 
Then, you need to know the number of new classes is meaningless. It is only for paper. Of course, add a new class will be easier than more. But in fact, If other papers that you want to compare use 1 classes, you need to add 1 class. You can imagine real worlds. Do you think it has any meaning? There are infinite classes you need to learn.

**Comment by Deexaa** [comment]
Thank you for your detailed answers.
As for the first question, I used to think the 2 datasets with different attribute may just be used as transfer learning.But now I realized maybe this kind of setting is just for easier classification.
As for the second question, the result of adding a new  per task is worse than adding two classes instead( I guess it attributes to appropriate hyper-parameters). Now I will decide the settings depending on the purpose(such as compare methods with other papers or real world setting).
Thanks again for your help! I wish you all the best:)
